src/quo2/components/wallet/network_routing/view.cljs

```diff
@@ -0,0 +1,182 @@
+  (assoc network
+         :amount-shared-value      (reanimated/use-shared-value amount)
+         :translate-x-shared-value (reanimated/use-shared-value 0)))
+
+(def ^:private get-negative-amount
+  (comp - reanimated/get-shared-value :amount-shared-value))
+
+(defn f-network-routing-bars
+  [_]
+  (let [selected-network-idx (reagent/atom nil)
+        press-locked?        (reagent/atom false)
+        lock-press           #(reset! press-locked? true)
+        unlock-press         #(reset! press-locked? false)
+        reset-state-values   #(reset! selected-network-idx nil)]
+    (fn [{:keys [networks total-width total-amount requesting-data? on-amount-selected]}]
+      (let [bar-opacity-shared-value (reanimated/use-shared-value 0)
+            network-bars             (map add-bar-shared-values networks)
+            amount->width            #(* % (/ total-width total-amount))
+            bars-widths-negative     (map #(-> % get-negative-amount amount->width)
+                                          network-bars)
+            last-bar-idx             (dec (count network-bars))]
+        (rn/use-effect
+         #(when (and (not requesting-data?) @selected-network-idx)
+            (let [bar (nth network-bars @selected-network-idx)]
+              (animation/hide-pressed-bar bar amount->width))
+            (animation/update-bar-values-and-reset-animations
+             {:new-network-values networks
+              :network-bars       network-bars
+              :amount->width      amount->width
+              :reset-values-fn    reset-state-values
+              :lock-press-fn      lock-press
+              :unlock-press-fn    unlock-press
+              :add-new-timeout    add-new-timeout}))
+         [requesting-data?])
+        [:<>
+         (doall
+          (for [[bar-idx bar] (map-indexed vector network-bars)
+                :let          [bar-max-width      (amount->width (:max-amount bar))
+                               bar-width          (-> (:amount-shared-value bar)
+                                                      (reanimated/get-shared-value)
+                                                      (amount->width))
+                               hide-division?     (or (= last-bar-idx bar-idx) @selected-network-idx)
+                               this-bar-selected? (= @selected-network-idx bar-idx)]]
+            ^{:key (str "network-bar-" bar-idx)}
+            [:f> f-network-bar
+             {:bar           bar
+              :max-width     bar-max-width
+              :total-width   total-width
+              :total-amount  total-amount
+              :bar-division? hide-division?
+              :on-top?       this-bar-selected?
+              :allow-press?  (and (or (not @selected-network-idx) this-bar-selected?)
+                                  (not requesting-data?)
+                                  (not @press-locked?))
+              :on-press      #(when-not @selected-network-idx
```

Comment:
Example of long function using `#()`. Using `(fn [] ...)` would clearly communicate the inner body doesn't care about any argument.

Reply:
done